File: docs/experiments/uri-fragment/javascript/uri_fragment_probe.mjs
const affectedBy = "http://open-services.net/ns/rm#affectedBy";
const trackedBy = "http://open-services.net/ns/rm#trackedBy";

const urlA = new URL(affectedBy);
const urlB = new URL(trackedBy);
const urlA2 = new URL(affectedBy);

console.log("JavaScript WHATWG URL");
console.log(`type=${urlA.constructor.name}`);
console.log(`a=${urlA.href} hash=${urlA.hash}`);
console.log(`b=${urlB.href} hash=${urlB.hash}`);
console.log(`a === b=${urlA === urlB}`);
console.log(`a.href === b.href=${urlA.href === urlB.href}`);
console.log(`a === a2=${urlA === urlA2}`);
console.log(`a.href === a2.href=${urlA.href === urlA2.href}`);
console.log(`Set<URL>.size=${new Set([urlA, urlB]).size}`);
console.log(`Set<URL> same lexical value size=${new Set([urlA, urlA2]).size}`);
console.log(`Set<string href>.size=${new Set([urlA.href, urlB.href]).size}`);

File: docs/experiments/uri-fragment/python/uri_fragment_probe.py
from urllib.parse import urlparse, urlsplit

affected_by = "http://open-services.net/ns/rm#affectedBy"
tracked_by = "http://open-services.net/ns/rm#trackedBy"

parse_a = urlparse(affected_by)
parse_b = urlparse(tracked_by)
split_a = urlsplit(affected_by)
split_b = urlsplit(tracked_by)

print("Python urllib.parse.urlparse")
print(f"type={type(parse_a).__module__}.{type(parse_a).__name__}")
print(f"a={parse_a.geturl()} fragment={parse_a.fragment}")
print(f"b={parse_b.geturl()} fragment={parse_b.fragment}")
print(f"a == b={parse_a == parse_b}")
print(f"hash(a)={hash(parse_a)} hash(b)={hash(parse_b)}")
print(f"set(ParseResult).count={len({parse_a, parse_b})}")

print()
print("Python urllib.parse.urlsplit")
print(f"type={type(split_a).__module__}.{type(split_a).__name__}")
print(f"a={split_a.geturl()} fragment={split_a.fragment}")
print(f"b={split_b.geturl()} fragment={split_b.fragment}")
print(f"a == b={split_a == split_b}")
print(f"hash(a)={hash(split_a)} hash(b)={hash(split_b)}")
print(f"set(SplitResult).count={len({split_a, split_b})}")
print(f"set(str).count={len({affected_by, tracked_by})}")

